sdk/keyvault/azsecrets/models.go

```diff
@@ -52,49 +35,76 @@ type Secret struct {
 	// The secret management attributes.
 	Properties *Properties `json:"attributes,omitempty"`
 
-	// The content type of the secret.
-	ContentType *string `json:"contentType,omitempty"`
-
 	// The secret id.
 	ID *string `json:"id,omitempty"`
 
-	// Application specific metadata in the form of key-value pairs.
-	Tags map[string]string `json:"tags,omitempty"`
-
 	// The secret value.
 	Value *string `json:"value,omitempty"`
+}
 
-	// READ-ONLY; If this is a secret backing a KV certificate, then this field specifies the corresponding key backing the KV certificate.
-	KeyID *string `json:"kid,omitempty" azure:"ro"`
-
-	// READ-ONLY; True if the secret's lifetime is managed by key vault. If this is a secret backing a certificate, then managed will be true.
-	Managed *bool `json:"managed,omitempty" azure:"ro"`
+func (s Secret) toGeneratedProperties() internal.SecretUpdateParameters {
+	var contentType *string
+	if s.Properties != nil && s.Properties.ContentType != nil {
+		contentType = s.Properties.ContentType
+	}
+	var tags map[string]*string
+	if s.Properties != nil && s.Properties.Tags != nil {
+		tags = convertToGeneratedMap(s.Properties.Tags)
+	}
+	return internal.SecretUpdateParameters{
+		ContentType:      contentType,
+		SecretAttributes: s.Properties.toGenerated(),
+		Tags:             tags,
+	}
 }
 
 // Properties - The secret management properties.
 type Properties struct {
+	// The content type of the secret.
+	ContentType *string `json:"contentType,omitempty"`
+
+	// READ-ONLY; Creation time in UTC.
+	CreatedOn *time.Time `json:"created,omitempty" azure:"ro"`
+
 	// Determines whether the object is enabled.
 	Enabled *bool `json:"enabled,omitempty"`
 
 	// Expiry date in UTC.
 	ExpiresOn *time.Time `json:"exp,omitempty"`
 
-	// Not before date in UTC.
-	NotBefore *time.Time `json:"nbf,omitempty"`
+	// READ-ONLY; True if the secret's lifetime is managed by key vault. If this is a secret backing a certificate, then managed
+	// will be true.
+	IsManaged *bool `json:"managed,omitempty" azure:"ro"`
 
-	// READ-ONLY; Creation time in UTC.
-	CreatedOn *time.Time `json:"created,omitempty" azure:"ro"`
+	// READ-ONLY; If this is a secret backing a KV certificate, then this field specifies the corresponding key backing the KV
+	// certificate.
+	KeyID *string `json:"kid,omitempty" azure:"ro"`
 
-	// READ-ONLY; Last updated time in UTC.
-	UpdatedOn *time.Time `json:"updated,omitempty" azure:"ro"`
+	// Not before date in UTC.
```

Comment:
Doc comments should begin with the field name
```suggestion
	// NotBefore is the secret's not before date in UTC.
```